hotel-booking-agent-autogen-agent-iam/README.md

```diff
@@ -0,0 +1,325 @@
+
+### Asgardeo Configuration (OAuth Provider)
+
+1. **Create an Asgardeo Account**:
+   - Sign up at [https://wso2.com/asgardeo/](https://wso2.com/asgardeo/)
+   - Create a new organization/tenant
+
+2. **Create and configure applications**:
+   1. Agent Application
+      - Navigate to **Applications** > **New Application**,
+      - Select **Standard-Based Application (OAuth/OIDC)** and choose **OAuth/OpenID Connect**.
+      - Click **create**
+      - Navigate to login flow tab and configure Agent authenticator in the first step.
+   
+   2. Hotel Booking Application 
+      - Navigate to **Applications** > **New Application**
+      - Select **Standard-Based Application (OAuth/OIDC)** and choose **OAuth/OpenID Connect**.
+      - Use the following values for the configuration.
+         - **Name**: Hotel Booking Assistant
+         - **Grant Types**: Check "Code", "Client Credentials" and "Refresh"
+         - **Callback URLs**: Add `http://localhost:8000/callback` (or your deployment URL)
+         - **Access Token Type**: JWT
+         - **Allowed Origins**: Add your frontend domain for CORS
+         - Navigate to **Roles** tab and set audience as "Organization"
+
+      After creating the app, copy the "Client ID" and "Client Secret". Use these in your environment variables (`ASGARDEO_CLIENT_ID` and `ASGARDEO_CLIENT_SECRET`)
+
+3. **Configure Hotel Booking API**:
+   - Navigate to "API Resources" and create a new API resource.
+   - Create custom scopes for the Hotel API:
+     - `read_hotels`: Permission to view hotel listings
+     - `read_rooms`: Permission to view room details
+     - `create_bookings`: Permission to make reservations
+     - Authorize this API to the hotel booking application created in step 2.
+
+4. **Role configuration**:
+   - Navigate to "Roles" and create an organization role "Guest".
+   - Assign the hotel API created in step 3 to the created role and add its scopes to the role.
+
+### Azure OpenAI Configuration
+
+1. **Create an Azure OpenAI Resource**:
+   - Log in to the [Azure Portal](https://portal.azure.com/)
+   - Search for "Azure OpenAI" and create a new resource
+   - Select a region where GPT models are available
+
+2. **Deploy a Model**:
+   - In your Azure OpenAI resource, navigate to "Deployments"
+   - Click "Create new deployment"
+   - Select model (e.g., "gpt-4o")
+   - Provide a deployment name (e.g., "hotel-assistant")
+
+3. **Get API Details**:
+   - Navigate to "Keys and Endpoint" in your Azure OpenAI resource
+   - Copy the "Endpoint" URL (like `https://your-resource.openai.azure.com/`)
+   - Copy one of the available keys
+
+4. **Set Environment Variables**:
+   - Use the endpoint URL for `AZURE_OPENAI_ENDPOINT`
+   - Use the keys copied for `AZURE_OPENAI_API_KEY`
+   - Use the deployment name for `AZURE_OPENAI_DEPLOYMENT_NAME`
+
+### Hotel API Configuration
+
+1. Deploy [Hotel API](https://github.com/nadheesh/agent_demo_hotel_api) using the given source in WSO2 Choreo.
```

Comment:
We need to move this to this same repo under a separate sub-folder such as sample-resources

Reply:
@shashimalcse will you be able to send a PR for this?